### PR TITLE
Fix guest user syncing with prefix and no Mail

### PR DESCRIPTION
**Pull request by asalkeld**
```release-note
Fix guest user syncing with prefix and no Mail
```
/cc @grantomation  @mjudeikis

### CODE REVIEW

**Comment by mjudeikis** [comment]
@Makdaam this clashes with your changes(?)

**Comment by asalkeld** [comment]
> @Makdaam this clashes with your changes(?)

This came in as an incident... @Makdaam are you also working on this?

**Comment by Makdaam** [comment]
I"m working on a different case. This doesn't clash with my case (having mail and userPrincipalName set, but using principal to log in.

**Comment by Makdaam** [comment]
Side note: I'll wait for this PR to merge, then rebase mine https://github.com/openshift/openshift-azure/pull/2128

**Comment by jim-minter** [comment]
/hold
while taking a look at this

File: pkg/controllers/customeradmin/group_mapping.go [hunk 1]
```diff
@@ -102,15 +103,16 @@ func reconcileUsers(log *logrus.Entry, ocpUserList []v1.User, AADUser graphrbac.
 			}
 		}
 	}
-	// This is optimistic code, trying to catch use-case 2
-	if AADUser.MailNickname != nil &&
-		AADUser.Mail != nil &&
-		strings.Contains(*AADUser.MailNickname, "#EXT#") {
+	// This is optimistic code, trying to catch use-case 2 and 2.b
+	if AADUser.MailNickname != nil && strings.Contains(*AADUser.MailNickname, "#EXT#") {
+		s := strings.Replace(*AADUser.MailNickname, "#EXT#", "", 1)
```
Comment (jim-minter) [inline pkg/controllers/customeradmin/group_mapping.go]: I think I'd prefer this instead:
```
var email string
if AADUser.Mail != nil {
	email = *AADUser.Mail
} else {
	s := strings.Replace(*AADUser.MailNickname, "#EXT#", "", 1)
	idx := strings.LastIndex(s, "_")
	email = s[:idx] + "@" + s[idx+1:]
}
```

Comment (jim-minter) [reply pkg/controllers/customeradmin/group_mapping.go]: is that workable? It retains the previous logic in this space.

**Comment by asalkeld** [comment]
@jim-minter ptal

**Pull request merged**